src/System.Composition.AttributedModel/src/System.Composition.AttributedModel.csproj

```diff
@@ -0,0 +1,66 @@
+    <TargetFrameworkProfile>Profile259</TargetFrameworkProfile>
+    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
+    <RestorePackages>true</RestorePackages>
+    <SccProjectName>SAK</SccProjectName>
+    <SccLocalPath>SAK</SccLocalPath>
+    <SccAuxPath>SAK</SccAuxPath>
+    <SccProvider>SAK</SccProvider>
+    <MinimumVisualStudioVersion>11.0</MinimumVisualStudioVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <OutputPath>bin\Debug\</OutputPath>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <DocumentationFile>bin\Debug\System.Composition.AttributedModel.XML</DocumentationFile>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <OutputPath>bin\Release\</OutputPath>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <DocumentationFile>bin\Release\System.Composition.AttributedModel.XML</DocumentationFile>
+  </PropertyGroup>
+  <PropertyGroup>
+    <ExternallyShipping>true</ExternallyShipping>
+  </PropertyGroup>
+  <ItemGroup>
+    <!-- A reference to the entire .NET Framework is automatically included -->
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="System\Composition\Convention\AttributedModelProvider.cs" />
+    <Compile Include="System\Composition\ExportAttribute.cs" />
+    <Compile Include="System\Composition\ExportMetadataAttribute.cs" />
+    <Compile Include="System\Composition\ImportAttribute.cs" />
+    <Compile Include="System\Composition\ImportingConstructorAttribute.cs" />
+    <Compile Include="System\Composition\ImportManyAttribute.cs" />
+    <Compile Include="System\Composition\ImportMetadataConstraintAttribute.cs" />
+    <Compile Include="System\Composition\MetadataAttributeAttribute.cs" />
+    <Compile Include="System\Composition\OnImportsSatisfiedAttribute.cs" />
+    <Compile Include="System\Composition\PartMetadataAttribute.cs" />
+    <Compile Include="System\Composition\PartNotDiscoverableAttribute.cs" />
+    <Compile Include="System\Composition\SharedAttribute.cs" />
+    <Compile Include="System\Composition\SharingBoundaryAttribute.cs" />
+  </ItemGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+       Other similar extension points exist, see Microsoft.Common.targets.
+  <Target Name="BeforeBuild">
+  </Target>
+  <Target Name="AfterBuild">
+  </Target>
+  -->
+</Project>
```

Comment:
> Does this matter at all? 

Yes, some example discussions of the topic:
- http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline
- http://stackoverflow.com/questions/5813311/no-newline-at-end-of-file
- http://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file
- https://robots.thoughtbot.com/no-newline-at-end-of-file

> Does this apply just to project files or also to source files? 

All text-based files.

> Should this be part of the code formatter?

That would help, but codeformatter doesn't process all files for which this is relevant.